tensorboard/webapp/feature_flag/views/feature_flag_modal_trigger_container.ts

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component} from '@angular/core';
+import {MatDialog, MatDialogRef} from '@angular/material/dialog';
+import {Store} from '@ngrx/store';
+import {State} from '../../app_state';
+import {featureFlagOverrideChanged} from '../actions/feature_flag_actions';
+import {
+  getFeatureFlags,
+  getShowFlagsEnabled,
+} from '../store/feature_flag_selectors';
+import {FeatureFlagPageContainer} from './feature_flag_page_container';
+
+@Component({
+  selector: 'feature-flag-modal-trigger',
+  template: ``,
+  styles: [],
+})
+export class FeatureFlagModalTriggerContainer {
+  readonly showFeatureFlags$ = this.store.select(getShowFlagsEnabled);
+  readonly featureFlags$ = this.store.select(getFeatureFlags);
+  private featureFlagsDialog?: MatDialogRef<FeatureFlagPageContainer>;
+
+  constructor(private readonly store: Store<State>, private dialog: MatDialog) {
+    this.showFeatureFlags$.subscribe((showFeatureFlags: boolean) => {
+      if (showFeatureFlags) {
+        this.featureFlagsDialog = this.dialog.open(FeatureFlagPageContainer);
+        this.featureFlagsDialog.afterClosed().subscribe(() => {
+          // By disabling the flag when the dialog is closed to prevent it from appearing again after the page is refreshed.
+          this.store.dispatch(
```

Comment:
Thinking about this after playing with the code locally: I think when the dialog closes the page should be refreshed.

My reasoning: This is the first time a user can change feature flag state AFTER the app has been loaded. This probably opens up a ton of subtle bugs that we never had to consider in the past. We shouldn't allow the user to play with the app in this state. Refreshing the page after the dialog is closed would put the app back into a proper state.

Reply:
Sure. Let's address in a followup PR. The key is that users should not be able to use the app after changing the flag state.